src/core/AutomationClip.cpp

```diff
@@ -1106,31 +1106,23 @@ void AutomationClip::generateTangents(timeMap::iterator it, int numToGenerate)
 {
 	QMutexLocker m(&m_clipMutex);
 
-	if( m_timeMap.size() < 2 && numToGenerate > 0 )
+	for (int i = 0; i < numToGenerate && it != m_timeMap.end(); ++i, ++it)
```

Comment:
I personally prefer we don't use the comma operator, since it made a bug hard to find awhile back. But if there isn't a choice somehow, then I don't mind.

Reply:
It's fairly idiomatic in the iteration expression of a for-loop when there are multiple iteration variables, and its arguments here aren't problematic. I agree in general though, and would avoid it if a semicolon would do the same job (which obviously isn't an option here). I think the other main place it's useful is in a fold expression when each element of the pack needs to be evaluated unconditionally.